src/main/java/it/pagopa/transactions/controllers/v2_1/TransactionsController.java

```diff
@@ -0,0 +1,278 @@
+                        .title("Bad request")
+                        .detail("Invalid request: %s".formatted(exception.getMessage())),
+                HttpStatus.BAD_REQUEST
+        );
+    }
+
+    @ExceptionHandler(
+        {
+                JWTTokenGenerationException.class
+        }
+    )
+    ResponseEntity<ProblemJsonDto> jwtTokenGenerationError(JWTTokenGenerationException exception) {
+        log.warn(exception.getMessage());
+        HttpStatus httpStatus = HttpStatus.INTERNAL_SERVER_ERROR;
+        return new ResponseEntity<>(
+                new ProblemJsonDto()
+                        .status(httpStatus.value())
+                        .title(httpStatus.getReasonPhrase())
+                        .detail("Internal server error: cannot generate JWT token"),
+                HttpStatus.INTERNAL_SERVER_ERROR
+        );
+    }
+
+    @ExceptionHandler(NodoErrorException.class)
+    public ResponseEntity<?> nodoErrorHandler(NodoErrorException e) {
+        String faultCode = e.getFaultCode();
+        ResponseEntity<?> response = nodeErrorToV2_1TransactionsResponseEntityMapping.getOrDefault(
+                faultCode,
+                new ResponseEntity<>(
+                        new GatewayFaultPaymentProblemJsonDto()
+                                .title("Bad gateway")
+                                .faultCodeCategory(
+                                        GatewayFaultPaymentProblemJsonDto.FaultCodeCategoryEnum.GENERIC_ERROR
+                                )
+                                .faultCodeDetail(faultCode),
+                        HttpStatus.BAD_GATEWAY
+                )
+        );
+
+        log.error(
+                "Nodo error processing request with fault code: [" + faultCode + "] mapped to http status code: [" +
+                        response.getStatusCode() + "]",
+                e
+        );
+        return response;
+    }
+
+    @ExceptionHandler(
+        {
+                InvalidNodoResponseException.class,
+        }
+    )
+    ResponseEntity<ProblemJsonDto> invalidNodoResponse(InvalidNodoResponseException exception) {
+        log.warn(exception.getMessage());
+        HttpStatus httpStatus = HttpStatus.BAD_GATEWAY;
+        return new ResponseEntity<>(
+                new ProblemJsonDto()
+                        .status(httpStatus.value())
+                        .title(httpStatus.getReasonPhrase())
+                        .detail(exception.getErrorDescription()),
+                httpStatus
+        );
+    }
+
+    @Warmup
+    public void postNewTransactionWarmupMethod() {
+        IntStream.range(0, 3).forEach(
+                idx -> {
+                    log.info("Performing warmup iteration: {}", idx);
+                    NewTransactionResponseDto newTransactionResponseDto = WebClient
+                            .create()
+                            .post()
+                            .uri("http://localhost:8080/v3/transactions")
```

Comment:
```suggestion
                            .uri("http://localhost:8080/v2.1/transactions")
```
this should be v 2.1, right?

Reply:
Yep, fixed with latest push